### ISSUE TITLE
Download after connection interrupted

### ISSUE THREAD

**Issue by karmac2015** [hgouveia/node-downloader-helper#4]
There is a way to auto resume downloading file after connection is interrupted ? I need a timeout option when connection is interrupted more than timeout so an error is triggered and if connection is established again so download will resume downloading.

**Comment by hgouveia** [comment]
Hello @karmac2015 , right now the library doesn't has this functionallity but is an interesting feature that i didn't thought about and i will add in the future

but for now, what you can do , is use the "error" event to retry with .resume or .start

```javascript
const url = 'http://YOUR_URL';
const destFolder  = 'some/path';
const dl = new DownloaderHelper(url, destFolder);

dl
    .on('error' => { dl.resume(); console.log('Download Retry')´; })
    .on('end', () => console.log('Download Completed'))
    .start();
```

could you try this and let me know if worked for you?

**Comment by karmac2015** [comment]
Ok I will try to use your concept but how can I set the timeout, so when error occurs after timeout is exceeded ?

**Comment by hgouveia** [comment]
> 
> 
> Ok I will try to use your concept but how can I set the timeout, so when error occurs after timeout is exceeded ?

this can be done in multiples way, you can get the time when the download starts and compare when the error triggers

```javascript
let startTime = 0;
const TIMEOUT = 5000; //5seg
const url = 'http://YOUR_URL';
const destFolder  = 'some/path';
const dl = new DownloaderHelper(url, destFolder);

dl
    .on('start', () => startTime = new Date())
    .on('error', (error) => { 
              const endTime = new Date();
              // probably you can check the error.status value
              if (endTime  - startTime => TIMEOUT){
               dl.resume(); 
               console.log('Download Retry');
              }
    })
    .on('end', () => console.log('Download Completed'))
    .start();
```

note : this code is just a example, might now work or has some typos

**Comment by karmac2015** [comment]
I tried your example and I set the timeout for:  15 * 60 * 1000 // 15 min, but when network is available again before that time download helper does not continue downloading file.
I don't know if you understand my problem, my network is unstable and I need to continue download file after the connection is corrupted and reconnected again

**Comment by hgouveia** [comment]
> 
> 
> I tried your example and I set the timeout for: 15 * 60 * 1000 // 15 min, but when network is available again before that time download helper does not continue downloading file.
> I don't know if you understand my problem, my network is unstable and I need to continue download file after the connection is corrupted and reconnected again


yeap, i got you issue, my question is when you got the timeout, do you get this error on the Error event?.
but basically the idea is the same if you can catch this error, the solution will be trigger the resume after that

about why if you trigger resume, it starts from the beginning, this could be that the URL where you are downloading the file doesnt support resume header 

btw i just created a Gitter Chat so we can discuss better about this, https://gitter.im/node-downloader-helper/Lobby

**Comment by hgouveia** [comment]
Closed, added in version `1.0.11`
